File: ParadigmaComp/15Questoes/python/questao1.py
# Classes e Objetos Básicos
class Carro:
    def __init__(self, marca, modelo, ano):
        self.marca = marca
        self.modelo = modelo
        self.ano = ano

    def detalhes(self):
        return f"Marca: {self.marca}, Modelo: {self.modelo}, Ano: {self.ano}"

# Instanciando 3 objetos
carro1 = Carro("Toyota", "Corolla", 2020)
carro2 = Carro("Honda", "Civic", 2018)
carro3 = Carro("Ford", "Mustang", 2021)

# Exibindo os detalhes
print(carro1.detalhes())
print(carro2.detalhes())
print(carro3.detalhes())

File: ParadigmaComp/15Questoes/golang/questao1.go
package main

import "fmt"
//  Classes e Objetos Básicos

type Carro struct {
	Marca  string
	Modelo string
	Ano    int
}

func (c Carro) Detalhes() string {
	return fmt.Sprintf("Marca: %s, Modelo: %s, Ano: %d", c.Marca, c.Modelo, c.Ano)
}

func main() {
	// Instanciando 3 objetos
	carro1 := Carro{"Toyota", "Corolla", 2020}
	carro2 := Carro{"Honda", "Civic", 2018}
	carro3 := Carro{"Ford", "Mustang", 2021}

	// Exibindo os detalhes
	fmt.Println(carro1.Detalhes())
	fmt.Println(carro2.Detalhes())
	fmt.Println(carro3.Detalhes())
}
